### PR TITLE
[Android] Remove legacy swap code

### PR DESCRIPTION
**Pull request by wchen342**
Resolves brave/brave-browser/issues/30542

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Run through the following parts of wallet and make sure nothing is broken:
1. Tap on any assets, and then tap either "Buy" or "Send".
2. Make sure we can go to the buy sites correctly, and Send Tx can be executed correctly.
3. On top wallet page, go to "Buy" and "Send" using the bottom sheet.
4. Also make sure both buy and send are working correctly.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java [hunk 1]
```diff
@@ -333,9 +333,8 @@ public List<NetworkInfo> stripNoBuySwapNetworks(
             List<NetworkInfo> networkInfos, BuySendSwapActivity.ActivityType type) {
         List<NetworkInfo> networkInfosFiltered = new ArrayList<>();
         for (NetworkInfo networkInfo : networkInfos) {
-            if (type == BuySendSwapActivity.ActivityType.BUY && Utils.allowBuy(networkInfo.chainId)
-                    || (type == BuySendSwapActivity.ActivityType.SWAP
-                            && Utils.allowSwap(networkInfo.chainId))) {
+            if (type == BuySendSwapActivity.ActivityType.BUY
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java:336]: nit: can rename `stripNoBuySwapNetworks` to `stripNoBuyNetworks`

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 2]
```diff
@@ -1661,59 +1090,20 @@ public void updateBuySendSwapAsset(
                     token.symbol, getResources().getDisplayMetrics().density, assetText, this, true,
                     (float) 0.5, true);
         }
-        if (buySend && (token.isErc721 || token.isNft)) {
+        if (token.isErc721 || token.isNft) {
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:1093]: nit: can remove `token.isErc721`, `token.isNft` is enough.

**Review by Pavneet-Sing — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java [hunk 3]
```diff
@@ -704,23 +387,17 @@ private void populateBalance(String balance, boolean from, int responseDecimals)
...
-            mConvertedFromBalance = fromToBalance;
-            if (mActivityType == ActivityType.SEND) mBtnBuySendSwap.setEnabled(fromToBalance != 0);
-        } else
-            mConvertedToBalance = fromToBalance;
-
-        if (mActivityType == ActivityType.SWAP) {
-            getSendSwapQuota(true, false);
-        }
+        mConvertedFromBalance = fromToBalance;
+        if (mActivityType == ActivityType.SEND) mBtnBuySendSwap.setEnabled(fromToBalance != 0);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java:398]: we can rename `mBtnBuySendSwap` to `mBtnBuySend`

**Review by SergeyZhukovsky — APPROVED** [review]
++. That's an awesome cleanup!

**Review by mkarolin — APPROVED** [review]
strings ++

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java (hunk 3 above)
Comment (wchen342) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java]: Fixed.

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java (hunk 2 above)
Comment (wchen342) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java]: Removed now.

File: android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java (hunk 1 above)
Comment (wchen342) [reply android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java:336]: Fixed.

**Pull request merged**